apps/emqx_rule_engine/src/emqx_rule_engine_schema.erl

```diff
@@ -40,13 +40,17 @@ tags() ->
 roots() -> ["rule_engine"].
 
 fields("rule_engine") ->
+    fields("rule_engine_api") ++
+        [
+            {rules,
+                ?HOCON(hoconsc:map("id", ?R_REF("rules")), #{
+                    desc => ?DESC("rule_engine_rules"), default => #{}
+                })}
+        ];
+fields("rule_engine_api") ->
```

Comment:
Since "rule_engine_api" is not really a type, it's better to extract the fields to a common function instead.
```
rule_engine_settings() ->
  [....
   ]
```